### PR TITLE
Added runtime dependency to fix the intermittent test failures

### PR DESCRIPTION
**Pull request by Daniel-Genkin**
Fixed the intermittent test failures caused by mono-config.json being loaded out of order

### CODE REVIEW

**Comment by Daniel-Genkin** [comment]
@kg Is this what you meant? Also, can you please add the necessary reviewers. I still don't have access to anything.

File: src/mono/sample/mbr/browser/runtime.js [hunk 1]
```diff
@@ -5,7 +5,9 @@ var Module = {
     config: null,
 
     preInit: async function() {
+        Module.addRunDependency('mono-config.json');
         Module.config = await MONO.mono_wasm_load_config("./mono-config.json");
+		Module.removeDependency('mono-config.json');
```
Comment (kg) [inline src/mono/sample/mbr/browser/runtime.js:10]: Indentation got mucked up here

**Comment by kg** [comment]
It might make sense to hoist all this complexity into mono_wasm_load_config, and introduce a 'mono_wasm_load_config_async' which adds the dependency, kicks off the async load, then removes the dependency. That way you don't have to touch all these files any time something changes, and it's easy for people to properly imitate the pattern.

**Comment by Daniel-Genkin** [comment]
Sorry, I was rushing. Haha this is straight from my nightmares.

File: src/mono/wasm/runtime/library_mono.js [hunk 2]
```diff
@@ -2365,12 +2365,14 @@ var MonoSupportLib = {
...
-		 * Loads the mono config file (typically called mono-config.json)
+		 * Loads the mono config file (typically called mono-config.json) asynchroniously
+		 * Note: the run dependencies are so emsdk actually awaits it in order.
 		 *
 		 * @param {string} configFilePath - relative path to the config file
 		 * @throws Will throw an error if the config file loading fails
 		 */
-		mono_wasm_load_config: async function (configFilePath) {		
+		mono_wasm_load_config: async function (configFilePath) {
+			Module.addRunDependency('mono-config.json');
```
Comment (lewing) [inline src/mono/wasm/runtime/library_mono.js:2375]:
```suggestion
			Module.addRunDependency(configFilePath);	
```

File: src/mono/wasm/runtime/library_mono.js [hunk 3]
```diff
@@ -2382,8 +2384,10 @@ var MonoSupportLib = {
 				} else { // shell or worker
 					config = JSON.parse(read(configFilePath)); // read is a v8 debugger command
 				}
+				Module.removeDependency('mono-config.json');
 				return config;
 			} catch(e) {
+				Module.removeDependency('mono-config.json');
 				return {message: "failed to load config file", error: e};
 			}
```
Comment (lewing) [inline src/mono/wasm/runtime/library_mono.js:2392]:
```suggestion
			} catch(e) {
				return {message: "failed to load config file", error: e};
			} finally {
				mono_wasm_load_config: async function (configFilePath);
			}
```

**Review by lewing — CHANGES_REQUESTED** [review]
pass configFilePath as the

**Review by lewing — APPROVED** [review]

**Comment by Daniel-Genkin** [comment]
Looks like the tests that were failing for Ankit and Katelyn now pass
https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-54587-merge-ee87aa6564394220aa/System.Net.Http.Enterprise.Tests/console.e71ce652.log?sv=2019-07-07&se=2021-07-13T02%3A56%3A40Z&sr=c&sp=rl&sig=cywykSQQsk2Ik7XAuj37PYFu5zTeNRzOpHBKwAlcHDQ%3D

**Comment by lewing** [comment]
This is JavaScript only change that cannot impact the other failing lanes.

**Pull request merged**